Mathlib/NumberTheory/NumberField/Embeddings.lean

```diff
@@ -1005,6 +1007,140 @@ lemma card_eq_card_isUnramifiedIn [NumberField k] [IsGalois k K] :
       Finset.card (Finset.univ.filter <| IsUnramifiedIn K (k := k))ᶜ * (finrank k K / 2) := by
   rw [← card_isUnramified, ← card_isUnramified_compl, Finset.card_add_card_compl]
 
+variable {K : Type*} [Field K] [NumberField K] (v : InfinitePlace K)
+
+noncomputable instance : Inhabited (InfinitePlace K) :=
+  ⟨Classical.choice (instNonemptyInfinitePlaceOfNumberField K)⟩
+
+/-- The normed field structure of a number field coming from the norm asociated to
+an infinite place. -/
+noncomputable def normedField : NormedField K :=
+  NormedField.induced _ _ v.embedding v.embedding.injective
+
+/-- The normed division ring structure of a number field coming from the norm associated
+to an infinite place. -/
+noncomputable def normedDivisionRing : NormedDivisionRing K :=
+  v.normedField.toNormedDivisionRing
+
+/-- The uniform structure of a number field at an infinite place. -/
+noncomputable def uniformSpace : UniformSpace K :=
+  v.normedField.toUniformSpace
+
+/-- The uniform additive group structure of a number field induced by an infinite place. -/
+theorem uniformAddGroup : @UniformAddGroup K v.uniformSpace _ :=
+  letI := v.normedField
+  inferInstance
+
+/-- The topology of a number field infuced by an infinite place. -/
+noncomputable def topologicalSpace : TopologicalSpace K :=
+  v.uniformSpace.toTopologicalSpace
+
+/-- The topological division ring structure of a number field induced by an infinite place. -/
+theorem topologicalDivisionRing :
+    @TopologicalDivisionRing K _ v.topologicalSpace :=
+  v.normedDivisionRing.to_topologicalDivisionRing
+
+/-- The topological ring structure of a number field induced by an infinite place. -/
+theorem topologicalRing : @TopologicalRing K v.topologicalSpace _ :=
+  @TopologicalDivisionRing.toTopologicalRing _ _ v.topologicalSpace
+    v.topologicalDivisionRing
```

Comment:
Are these really needed?

Reply:
I had these in because I couldn't get the type class inferencing to work, but the `letI` approach resolves this nicely and tidies it up a lot, thanks!